### PR TITLE
ref(js): Remove OrganizationsActions

### PR DESCRIPTION
**Pull request by evanpurkhiser**

### CODE REVIEW

File: static/app/actionCreators/organizations.tsx [hunk 1]
```diff
@@ -105,14 +104,15 @@ export function removeAndRedirectToRemainingOrganization(
  * Set active organization
  */
 export function setActiveOrganization(org: Organization) {
-  OrganizationsActions.setActive(org);
+  GuideStore.setActiveOrganization(org);
+  LatestContextStore.onSetActiveOrganization(org);
```
Comment (evanpurkhiser) [inline static/app/actionCreators/organizations.tsx:108]: Only these two things were listening to this action

File: static/app/actionCreators/organizations.tsx [hunk 2]
```diff
@@ -121,7 +121,8 @@ export function changeOrganizationSlug(
  * Accepts a partial organization as it will merge will existing organization
  */
 export function updateOrganization(org: Partial<Organization>) {
-  OrganizationsActions.update(org);
+  OrganizationsStore.onUpdate(org);
+  LatestContextStore.onUpdateOrganization(org);
```
Comment (evanpurkhiser) [inline static/app/actionCreators/organizations.tsx:125]: These were the two things listening to this action

File: static/app/stores/guideStore.tsx [hunk 3]
```diff
@@ -84,31 +80,27 @@ interface GuideStoreDefinition extends CommonStoreDefinition<GuideStoreState> {
   nextStep(): void;
   recordCue(guide: string): void;
   registerAnchor(target: string): void;
+  setActiveOrganization(data: Organization): void;
   setForceHide(forceHide: boolean): void;
   state: GuideStoreState;
+  teardown(): void;
```
Comment (evanpurkhiser) [inline static/app/stores/guideStore.tsx:86]: Still need the teardown for cleaning up the window event listeners

File: static/app/stores/guideStore.tsx [hunk 4]
```diff
@@ -125,7 +117,7 @@ const storeConfig: GuideStoreDefinition = {
     this.updateCurrentGuide();
   },
 
-  onSetActiveOrganization(data) {
+  setActiveOrganization(data: Organization) {
```
Comment (evanpurkhiser) [inline static/app/stores/guideStore.tsx:120]: This is just consistent with the rest of the store's interface

File: static/app/actionCreators/organizations.tsx [hunk 5]
```diff
@@ -75,15 +76,13 @@ export function remove(api: Client, {successMessage, errorMessage, orgId}: Remov
...
     .then(() => {
-      OrganizationsActions.removeSuccess(orgId);
+      OrganizationsStore.onRemoveSuccess(orgId);
 
       if (successMessage) {
         addSuccessMessage(successMessage);
       }
     })
     .catch(() => {
-      OrganizationsActions.removeError();
```
Comment (evanpurkhiser) [inline static/app/actionCreators/organizations.tsx:85]: This wasn't listened to at all

**Review by malwilley — APPROVED** [review]

**Comment by evanpurkhiser** [comment]
Have to fix some TS errors first

**Pull request merged**